src/Sql/Sql/Instance Pools/Cmdlet/GetAzureSqlInstancePool.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.Sql.Instance_Pools.Model;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+
+namespace Microsoft.Azure.Commands.Sql.Instance_Pools.Cmdlet
+{
+    /// <summary>
+    /// Defines the Get-AzRmSqlInstancePool cmdlet
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "SqlInstancePool")]
+    [OutputType(typeof(AzureSqlInstancePoolModel))]
+    public class GetAzureSqlInstancePool : InstancePoolCmdletBase
+    {
+        /// <summary>
+        /// Parameter setss
+        /// </summary>
+        private const string ListBySubOrResourceGroupParameterSet = "ListBySubOrResourceGroupParameterSet";
+        private const string GetInstancePoolDefaultParameterSet = "ListByInstancePoolDefaultsParameterSet";
+        private const string GetInstancePoolByInstancePoolResourceIdentifierParameterSet = "GetInstancePoolByInstancePoolResourceIdentifierParameterSet";
+
+        /// <summary>
+        /// Gets or sets the resource group name
+        /// </summary>
+        [Parameter(
+            ParameterSetName = ListBySubOrResourceGroupParameterSet,
+            Mandatory = false,
+            Position = 0,
+            HelpMessage = "The resource group name.")]
+        [Parameter(
+            ParameterSetName = GetInstancePoolDefaultParameterSet,
+            Mandatory = true,
+            Position = 1,
+            HelpMessage = "The resource group name.")]
+        [ResourceGroupCompleter]
+        public override string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the instance pool name
+        /// </summary>
+        [Parameter(
+            ParameterSetName = GetInstancePoolDefaultParameterSet,
+            Position = 1,
+            Mandatory = true,
+            HelpMessage = "The instance pool name.")]
+        [Alias("InstancePoolName")]
+        [ResourceNameCompleter("Microsoft.Sql/instancePools", "ResourceGroupName")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Gets or sets the instance pool resource identifier.s
+        /// </summary>
+        [Parameter(
+            ParameterSetName = GetInstancePoolByInstancePoolResourceIdentifierParameterSet,
+            Position = 0,
+            Mandatory = true,
+            HelpMessage = "The instance pool resource identifier.")]
+        public string ResourceId { get; set; }
```

Comment:
@johnpaulkee please make sure that this parameter has the property `ValueFromPipelineByPropertyName = true` to enable piping

Reply:
Acked.